test/test_mixed_null_assertions.cpp

```diff
@@ -78,3 +78,258 @@ TEST(List_Mixed_do_insert)
+            found.push_back(pos);
+        });
+        CHECK(check_results({2}));
+    }
+
+    {
+        // find null or find unresolved link diverge, different objects should be returned
+        auto index = list.find_any(realm::null());
+        CHECK(index == 0);
+        index = list.find_first(obj1);
+        CHECK(index == 2);
+        // but both should look like nulls
+        CHECK(list.is_null(0));
+        CHECK(list.is_null(2));
+    }
+
+    {
+        std::vector<size_t> indices{0, 1, 2};
+        list.sort(indices);
+        CHECK(indices.size() == 3);
+        CHECK(indices.at(0) == 0);
+        CHECK(indices.at(1) == 2);
+        CHECK(indices.at(2) == 1);
+        CHECK(list.is_null(indices[0]));
+        CHECK(list.is_null(indices[1]));
+        CHECK(!list.is_null(indices[2]));
+    }
+
+    {
+        std::vector<size_t> indices{0, 1, 2};
+        list.distinct(indices);
+        CHECK(indices.size() == 2);
+        CHECK(indices.at(0) == 0);
+        CHECK(indices.at(1) == 1);
+        CHECK(list.is_null(indices[0]));
+        CHECK(!list.is_null(indices[1]));
+        CHECK(list.find_any(realm::null()) == 0);
+    }
+
+    {
+        list.remove(0);
+        CHECK(list.find_any(obj1) == 1);
+        list.remove(1);
+        CHECK(list.find_any(realm::null()) == npos);
+        CHECK(list.size() == 1);
+    }
+
+    {
+        Group g;
+        auto t = g.add_table("foo");
+        t->add_column_list(type_Mixed, "mixeds");
+        auto obj = t->create_object();
+        auto obj1 = t->create_object();
+        auto list = obj.get_list<Mixed>("mixeds");
+
+        list.insert(0, obj1);
+        list.insert_null(1);
+        obj1.invalidate();
+
+        auto index_any = list.find_any(realm::null());
+        auto index_first = list.find_first(realm::null());
+        CHECK_EQUAL(index_any, 0);
+        CHECK_EQUAL(index_first, 0);
+    }
+
+    {
+        Group g;
+        auto t = g.add_table("foo");
+        t->add_column_list(type_Mixed, "mixeds");
+        auto obj = t->create_object();
+        auto obj1 = t->create_object();
+        auto list = obj.get_list<Mixed>("mixeds");
+
+        list.insert(0, obj1);
+        obj1.invalidate();
+        auto index_any = list.find_any(realm::null());
+        auto index_first = list.find_first(realm::null());
+        CHECK_EQUAL(index_any, 0);
+        CHECK_EQUAL(index_first, 0);
+    }
+}
+
+TEST(Mixed_Set_unresolved_links)
+{
+    Group g;
+
+    auto t = g.add_table("foo");
+    t->add_column_set(type_Mixed, "mixeds");
+    auto obj = t->create_object();
+    auto obj1 = t->create_object();
+    auto obj2 = t->create_object();
+    auto set = obj.get_set<Mixed>("mixeds");
+    auto [it, success] = set.insert(Mixed{obj1});
+    obj1.invalidate();
+
+    CHECK(success);
+    auto [it1, success1] = set.insert(Mixed{"test"});
+    CHECK(success1);
+
+    {
+        // null can be inserted in the set
+        CHECK(set.size() == 2);
```

Comment:
I would prefer to use `CHECK_EQUAL(set.size(), 2);` If the check fails, you will immediately see the difference. Should be changed in places where applicable.